### PR TITLE
[RHELC-875] Porting out optparse to argparse

### PR DESCRIPTION
**Pull request by r0x0d**
This is the initial work before we deep dive into making changes for the pre conversion framework, as we will introduce subcommands there, we need to at least, have our current CLI module to be using argparse for those features.

The current optparse usage doesn't support subcommands natively.

Jira Issues: [RHELC-875](https://issues.redhat.com/browse/RHELC-875)

Checklist

- [ ] PR has been tested manually in a VM (either author or reviewer)
- [ ] Jira issue has been made public if possible
- [ ] `[RHELC-]` is part of the PR title 
- [ ] GitHub label has been added to help with Release notes 
- [ ] PR title explains the change from the user's point of view
- [ ] Code and tests are documented properly
- [ ] The commits are squashed to as few commits as possible (without losing data) 
- [ ] When merged: Jira issue has been updated to `Release Pending` if relevant

### CODE REVIEW

File: convert2rhel/toolopts.py [hunk 1]
```diff
@@ -79,7 +79,7 @@ def __init__(self):
     def _get_argparser():
         usage = (
             "\n"
-            "  convert2rhel [-h]\n"
+            "  convert2rhel [-h]<n>"
```
Comment (abadger) [inline convert2rhel/toolopts.py:82]: Is this correct?

File: convert2rhel/toolopts.py [hunk 2]
```diff
@@ -138,28 +139,68 @@ def _register_options(self):
...
+        group.add_argument(
+            "--disable-submgr",
+            action="store_true",
+            help="Replaced by --no-rhsm. Both options have the same effect.",
+        )
+        group.add_argument(
+            "--no-rhsm",
+            action="store_true",
+            help="Do not use the subscription-manager, use custom repositories instead. See --enablerepo/--disablerepo"
+            " options. Without this option, the subscription-manager is used to access RHEL repositories by default."
```
Comment (abadger) [inline convert2rhel/toolopts.py:186]:
```suggestion
            " options. Without this option, subscription-manager is used to access RHEL repositories by default."
```

**Review by abadger — CHANGES_REQUESTED** [review]
A few minor changes to make to the help strings but otherwise this looks good.

File: convert2rhel/toolopts.py (hunk 1 above)
Comment (r0x0d) [reply convert2rhel/toolopts.py:82]: Oh, I must have changed it by mistake while tyring out a few things. Thanks for spotting that out! 

```suggestion
            "  convert2rhel [-h]\n"
```

**Review by abadger — APPROVED** [review]

**Pull request merged**